control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllersv2
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+)
+
+// MeshGatewayController reconciles a MeshGateway object.
+type MeshGatewayController struct {
+	client.Client
+	Log                  logr.Logger
+	Scheme               *runtime.Scheme
+	MeshConfigController *MeshConfigController
+}
+
+// +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=meshgateway,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=meshgateway/status,verbs=get;update;patch
```

Comment:
I'm not sure if these actually do anything anymore, but all of the other controllers in this package have them.
They imply they'd be used to generate the `ClusterRole` for the connect-inject-controller, but the `ClusterRole` doesn't appear to be generated, and I can't find any command that would do so either.

Maybe @thisisnotashwin knows?

Reply:
yeah.. so this does create/updates the controller in the `config` folder under `control-plane` but for the most part, the RBAC is self managed. my only argument for continuing to keep it is to sanity check if we have in fact updated the RBAC for connect-inject correctly, but there are other ways we can tackle that. 

my only request would be that if we do delete those two lines, can we do there for all the V2 CRDs as well to avoid confusion for anybody trying to reference these files?

Reply:
I've regenerated the role in the `config` folder that @thisisnotashwin referenced

Reply:
thanks so much Nathan! Ill give this PR a look shortly.. sorry have been pulled in many directions the last couple days